src/rvsdg/from_cfg.rs

```diff
@@ -331,32 +399,14 @@ impl<'a> RvsdgBuilder<'a> {
             input_vars.push(var);
         }
 
-        let mut next = None;
         for (_, succ) in succs {
             // First, make sure that all inputs are correctly bound to inputs to the block.
             for (i, var) in input_vars.iter().copied().enumerate() {
                 self.store.insert(var, Operand::Arg(i));
             }
             // Loop until we reach a join point.
             let mut curr = succ;
-            loop {
-                // Join points are nodes with more than one predecessor
-                // (excluding loop back-edges).
-                if self
-                    .cfg
-                    .graph
-                    .neighbors_directed(curr, Direction::Incoming)
-                    .filter(|neigh| {
-                        let Some(mut dom) = self.dom.dominators(*neigh) else {
-                            return true;
-                        };
-                        !dom.any(|n| n == curr)
-                    })
-                    .nth(1)
-                    .is_some()
-                {
-                    break;
-                }
+            while curr != join_point {
                 curr = self.try_loop(curr)?.unwrap();
```

Comment:
What does it mean for try_loop or try_branch to return None?

Reply:
It happens if you reach the exit block early.